### PR TITLE
adding chapter 2 exercises

### PR DESCRIPTION
**Pull request by clkwrkunvrs**

### CODE REVIEW

File: loopingATriangle.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+var ptrn = '#';
```
Comment (CelestineKao) [inline loopingATriangle.js]: generally speaking, bias towards using whole words for variable naming. there should also be an additional space between this line and the `for` loop.

File: chessBoard.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const size = 12;
+let hashString1 = (" #").repeat(size/2);
+let hashString2 = ("# ").repeat(size/2);
+
+for (let i = 0; i < size; i++) {
+    i % 2 === 0 ? console.log(hashString1) : console.log(hashString2);
```
Comment (CelestineKao) [inline chessBoard.js]: the norm for JS is to have two spaces per level of indentation, can you fix the indentation for all the files to follow the norm please?

File: fizzBuzz.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+for(let i = 1; i <= 100; i++) {
+    if (i % 3 === 0 && i % 5 === 0) {
+        console.log("FizzBuzz");
+    } else if (i % 3 === 0) {
+        console.log("Fizz");
+    } else if (i % 5 === 0) {
+        console.log("Buzz");
```
Comment (CelestineKao) [inline fizzBuzz.js]: this works fine, but can you think of a way to make this more terse?

File: chessBoard.js (hunk 2 above)
Comment (CelestineKao) [inline chessBoard.js]: i don't believe this works for odd numbers. 

could you add some basic interview-style tests to your files? something like:
```js
const chessBoard = (size) => {
  // code here
};

# Tests
chessBoard(0)
chessBoard(1)
chessBoard(8)
chessBoard(5)
```

for interviews, it's fine to not actually use a testing framework like `jest` or whatever, and instead just call the functions with several different inputs testing edge cases and whatnot and then just doing a visual comparison of outputs. the goal here is more about making sure you're testing your code in some way for correctness and that you're exercising edge cases.

**Review by CelestineKao — CHANGES_REQUESTED** [review]
this already looks better than last time! could you also put all the problems for a given chapter into a folder, or adopt some file naming convention so i can easily determine which chapter and problem i should refer to? something like `<chapter>_<problem number>_<descriptive name>.js`, ie`1_1_loopingTriangle.js`, `1_2_fizzBuzz.js`, etc?

File: ch2/fizzBuzz.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+for(let i = 1; i <= 100; i++) {
+  console.log(
+    !(i % 3) && !(i % 5) ? "FizzBuzz" :
+    !(i % 3) ? "Fizz" :
+    !(i % 5) ? "Buzz" :
+    i
```
Comment (CelestineKao) [inline ch2/fizzBuzz.js:6]: yea this works, though maybe more terse than optimal since chained ternaries like this can become hard to read. i was more referring to removing the separate case for the string "FizzBuzz", ie:

```js
for(let i = 1; i <= 100; i++) {
  let output = "";
  if (output % 3) output += "Fizz";
  if (output % 5) output += "Buzz";
  console.log(output === "" ? i :  output);
}
```

no need to change, but moving forward, try to avoid chained ternary expressions!

File: ch2/loopingATriangle.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+let pattern = '#';
+
+for(let i = 0; i < 7; i++) {
+	console.log(pattern);
+	pattern += '#';
```
Comment (CelestineKao) [inline ch2/loopingATriangle.js:5]: indentation needs to be fixed here, see https://github.com/clkwrkunvrs/c2bc/pull/1#discussion_r1534954950

File: ch2/fizzBuzz.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+    );
+}
```
Comment (CelestineKao) [inline ch2/fizzBuzz.js:8]: your code works, so i won't ask you to add tests, since i don't want to bog you down, but moving forward, please add tests!

**Review by CelestineKao — CHANGES_REQUESTED** [review]

File: ch2/fizzBuzz.js (hunk 4 above)
Comment (clkwrkunvrs) [reply ch2/fizzBuzz.js:6]: Well that’s pretty clever. Ok noted. lol you said more terse. This did achieve that particular objective.

File: ch2/fizzBuzz.js (hunk 6 above)
Comment (clkwrkunvrs) [reply ch2/fizzBuzz.js:8]: Wilco. Acknowledged
